File: java/src/main/java/ddp/engine/EngineError.java
package ddp.engine;

/**
 * Thrown when the ddp binary fails (invalid key, parse error, etc.).
 */
public class EngineError extends RuntimeException {
    private final String stderr;

    public EngineError(String message) {
        this(message, "");
    }

    public EngineError(String message, String stderr) {
        super(message);
        this.stderr = stderr != null ? stderr : "";
    }

    public String getStderr() {
        return stderr;
    }
}

File: csharp/DDPEngine/EngineError.cs
namespace DDPEngine;

/// <summary>Thrown when the ddp binary fails (invalid key, parse error, etc.).</summary>
public class EngineError : Exception
{
    /// <summary>Stderr output from the ddp binary.</summary>
    public string Stderr { get; }

    public EngineError(string message, string stderr = "")
        : base(message)
    {
        Stderr = stderr;
    }
}
